### DISCUSSION TITLE
Compile with pre-defined script.

### DISCUSSION THREAD

**Discussion by TroeperDeTroep** [arendst/Tasmota#15873] [Q&A] [answered]
I’m unable to find an options to compile a firmware including a predefined script.
For rules, you have option "#define USER_RULE1” in “my_user_config.h” file.

After weeks of searching and reading (including discord), I was unable to achieve this ):

**Comment by ascillato** [comment, marked as answer]
Hi,

That is already supported.

Just add what you need in the _**user_config_override.h**_ file as the following example:

```cpp
#define START_SCRIPT_FROM_BOOT
#define USER_BACKLOG "script 1"
#define PRECONFIGURED_SCRIPT ">D\n>B\nsmlj=0\n;->sensor53 r\n>R\nsmlj=0\n>S\nif upsecs>30\nthen\nsmlj=1\nendif\n>M 1\n+1,26,mN2,16,9600,Solar,32,10,01040008,0104000B,0104000E\n1,=h------------BATTERY------------\n1,01040SSss@i0:327.68,Voltage,V,Voltage,3\n1,01040SSss@i1:413.94,Current,A,Current,3\n1,01040ssxx@i2:100,Temp,°C,Temperature,2\n#"
```

**Reply by TroeperDeTroep** [reply]
Thank you so much for your elaborated explanation!!
This works like a charm.

Much appreciated.

**Reply by Elektrofix-OL** [reply]
Great, but how do I replace the " to define texts?

**Reply by barbudor** [reply]
```
#define EXAMPLE "Put in \"quotes\""
```
